### ISSUE TITLE
[BUG] Sim bursty oscillation bug

### ISSUE THREAD

**Issue by ryanhammonds** [neurodsp-tools/neurodsp#271]
```python
import numpy as np
from neurodsp.sim import sim_bursty_oscillation

np.random.seed(1)

n_seconds = 10
fs = 1000

sim_bursty_oscillation(n_seconds, fs, 15)
```

This traces into updates related to `make_bursts`.

**Comment by TomDonoghue** [comment]
I had a quick look into this, since I worked on these burst functions recently enough to mostly remember how it all works. 

The issue is this part of `make_bursts`:
```
for sig_ind, is_osc in zip(range(0, n_samples, n_samples_cycle), is_oscillating):
    if is_osc:
        burst_sig[sig_ind:sig_ind+n_samples_cycle] = cycle
```

Basically, this bug will occur if a cycle has a number of samples (`n_cycles_cycle`) that doesn't divide evenly into the total number of cycles for the signal (`n_cycles`), and if there is a cycle set to be added as the last cycle. This leads to the index error that gets raised, since the slice into burst_sig ends up being too small for the length of `cycle`. 
The reason it's a bit random if/when it occurs is that it's not-deterministic if there is a cycle set for the last possible cycle position, and so this will randomly work if the last cycle is False (in `is_oscillating`). 

Fixed proposed in #273
